### PR TITLE
Send documents.

### PR DESCRIPTION
**Pull request by guilleripa**
#### :link: Trello board reference:

- [Enviar archivos a través del chat widget](https://trello.com/c/9WzGFwtp/137-enviar-archivos-a-trav%C3%A9s-del-chat-widget)

---

#### :pencil2: Description:

By clicking on the paperclip icon on the chat input a window pops app letting a client send a document, of up to 5MB, to their support.
The document appears inline in the chat along with all other messages.

It also lets a user download a previously sent document.

---

#### :movie_camera: Screen record:

<p align="center">

</p>

---

### CODE REVIEW

File: src/components/Message/styles.less [hunk 1]
```diff
@@ -2,17 +2,32 @@
...
-  display: table;
+  display: flex;
   margin-bottom: 5px;
   margin-top: 5px;
   max-width: 235px;
   padding: 10px;
+  position: relative;
   text-align: left;
+  word-break: break-all;
   word-wrap: break-word;
```
Comment (agupirotto) [inline src/components/Message/styles.less]: Why did you add this? Are they not exclusive to each other?

Comment (guilleripa) [reply src/components/Message/styles.less]: `word-wrap: break-word` breaks a paragraph in between words, the other breaks a single word if it's to long for a single paragraph.

Comment (agupirotto) [reply src/components/Message/styles.less]: Sorry, I thought you were overwriting the CSS rules. I just realized that one is `word-wrap` and the other `word-break`.

`word-break: break-all`  it is used to break the words at any character 
Screenshot from 2019-11-02 23-55-47

I prefer this other behavior, `word-break: break-word`
Screenshot from 2019-11-02 23-55-38

Comment (guilleripa) [reply src/components/Message/styles.less]: I also like that! thanks for the protip!

File: src/components/Message/index.jsx [hunk 2]
```diff
@@ -1,23 +1,36 @@
...
-      unwrapDisallowed
+const Message = ({ message }) => {
+  let displayedMessage = undefined
+  if (message.attachment !== undefined) {
+    displayedMessage = (
+      <Tooltip title={`Download: ` + message.body}>
+        <Button
+          type="link"
+          icon="download"
+          onClick={() => download(message.attachment, message.body)}
```
Comment (eb16) [inline src/components/Message/index.jsx]: This `download` isn't needed. Everything that Ant doesn't expose in its interface, will send props down in the component as `{ ...rest }`. So, you can perfectly do this and will work (I've already tried it):

```
<Button
          type="link"
          icon="download"
          href={message.attachment}
          download={message.body}
        >
          {message.body}
</Button>
```

Underneath this button works as an `a` too, so your problem is resolved 😄

Comment (guilleripa) [reply src/components/Message/index.jsx]: Nice lookup into the documentation. Will definitely implement it this way!

File: src/context/messages/index.jsx [hunk 3]
```diff
@@ -169,12 +170,44 @@ const MessagesContextProvider = ({ children }) => {
     signalMessageQueue(state, chat, dispatch)
   }
 
+  const toBase64 = file =>
+    new Promise((resolve, reject) => {
+      const reader = new FileReader()
+      reader.readAsDataURL(file)
+      reader.onload = () => resolve(reader.result)
```
[the remark this answers is not in the document]
Comment (guilleripa) [reply src/context/messages/index.jsx]: We didn't define any restrictions for this iteration

[the remark this answers is not in the document]
Comment (guilleripa) [reply src/context/messages/index.jsx]: Update! We now allow all the types supported by carrierwave!
